pilot/pkg/xds/endpoint_builder.go

```diff
@@ -253,13 +257,19 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 	keys := shards.Keys()
 	// The shards are updated independently, now need to filter and merge for this cluster
 	for _, shardKey := range keys {
-		endpoints := shards.Shards[shardKey]
-		// If the downstream service is configured as cluster-local, only include endpoints that
-		// reside in the same cluster.
-		if isClusterLocal && (shardKey.Cluster != b.clusterID) {
-			continue
+		if shardKey.Cluster != b.clusterID {
+			// If the downstream service is configured as cluster-local, only include endpoints that
```

Comment:
One thing I wonder...

Say we add it to AugmentLabels. Now users can do "locality failover" with "node local" as an option (this is good, we have a FR for this). But that is different from "ONLY send node local", since it has the spillover mechanism.

I wonder if it makes sense to add a mode to DR that there is no spillover at all. Then we can implement this in terms of that option?

We can do this short term for now though, just an idea.

Reply:
> Now users can do "locality failover" with "node local" as an option (this is good, we have a FR for this). But that is different from "ONLY send node local", since it has the spillover mechanism.

If users don't want spillover they could just set `internalTrafficPolicy: Local`?

> I wonder if it makes sense to add a mode to DR that there is no spillover at all. Then we can implement this in terms of that option?

That may be related to the [overprovisioning factor](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/overprovisioning.html?highlight=overpr), there was [an related issue](https://github.com/istio/istio/issues/39445)
But IMO DR should not be coupled with this feature

Reply:
BTW I have tested failover priorities with a DR like this, seems to work fine
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: dr
spec:
  host: helloworld
  trafficPolicy:
    outlierDetection:
      consecutive5xxErrors: 20
      interval: 5m
      baseEjectionTime: 15m
    loadBalancer:
      localityLbSetting:
        failoverPriority:
          - "kubernetes.io/hostname"
```

Reply:
> If users don't want spillover they could just set internalTrafficPolicy: Local? The proxy will only see node local endpoints


That only works for "node local" though. My thought is to make it work for arbitrary localtiy config (following failoverPriority). Then we can implement internalTrafficPolicy in the terms of `localityLbSetting`

Reply:
I see, off the top of my head "no spillover" would probably be implemented by filtering endpoints like this pr